src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,119 @@
+package mate.academy.dao;
+
+import java.math.BigDecimal;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.connection.ConnectionUtil;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.Book;
+
+@Dao
+public class BookDaoImpl implements BookDao {
+    private static final String TITLE_COLUMN = "title";
+    private static final String PRICE_COLUMN = "price";
+    private static final String INSERT_BOOK_SQL = "INSERT INTO book (title, price) VALUES (?, ?)";
+    private static final String FIND_BY_ID_SQL = "SELECT * FROM book WHERE id = ?";
+    private static final String FIND_ALL_SQL = "SELECT * FROM book";
+    private static final String UPDATE_BOOK_SQL =
+            "UPDATE book SET title = ?, price = ? WHERE id = ?";
+    private static final String DELETE_BOOK_SQL = "DELETE FROM book WHERE id = ?";
```

Comment:
The SQL keywords in your queries should be in uppercase to adhere to SQL style best practices. For example, use 'INSERT INTO' instead of 'insert into'.